java-api-client/src/test/java/com/idibon/api/model/TaskTest.java

```diff
@@ -57,4 +59,76 @@
+            "\"labels\":[{\"name\":\"American Short Hair\"},{\"name\":\"Siamese\"}," +
+            "{\"name\":\"Persian\"},{\"name\":\"Burmese\"},{\"name\":\"Siberian\"}," +
+            "{\"name\":\"Balinese\"},{\"name\":\"Russian Blue\"},{\"name\":\"Maine Coon\"}]," +
+            "\"features\":[{\"uuid\":\"00000000-0000-0000-0000-000000000001\"," +
+            "\"name\":\"ClarabridgeRule\",\"parameters\":{\"label_rules\":" +
+            "{\"American Short Hair\":[[\"\",\"\",\"\",\"\"]],\"Siamese\":[[\"\",\"\",\"\",\"\"]]," +
+            "\"Persian\":[[\"\",\"\",\"\",\"\"]]," + 
+            "\"Burmese\":[[\"\",\"\",\"\",\"\"]],\"Siberian\":[[\"\",\"\",\"\",\"\"]]," +
+            "\"Balinese\":[[\"\",\"\",\"\",\"\"]],\"Russian Blue\":[[\"\",\"\",\"\",\"\"]]," +
+            "\"Maine Coon\":[[\"\",\"\",\"\",\"\"]]}},\"is_active\":true}]}}";
+        JsonObject taskJson = Json.createReader(new StringReader(json)).readObject();
+        Collection mockCollection = Collection.instance(null, "C");
+        Task mockTask = Task.instance(mockCollection, taskJson);
+
+        /* Creating fake documents. This could really be anything, since it won't be used in 
+         * an actual prediction. Since there are no rules, it is a trivial accept */
+        Document mockDocument = Document.instance(mockCollection, "Cats are similar in anatomy to the " +
+            "other felids, with strong, flexible bodies, quick reflexes, sharp retractable claws, " + 
+            "and teeth adapted to killing small prey.");
+        List<DocumentContent> docs = new ArrayList();
+        for (int i = 0; i < 2; i++) {
+            docs.add(mockDocument);
+        }
+        
+        // Ensure that the top-level prediction is 1.0
+        PredictionIterable<DocumentPrediction> prediction = mockTask.classifications(docs);
+        JsonArray predictionJson = prediction.iterator().next().right.getJson();
+        assert(predictionJson.getJsonObject(0).get("confidence").toString().equals("1.0"));
+        
+        // Ensure that all labels are predicted as 1.0
+        JsonObject labels = predictionJson.getJsonObject(0).getJsonObject("classes");
+        for (String label : labels.keySet()) {
+            assert(labels.get(label).toString().equals("1.0"));
+        }
+        
+        // Ensure that there are no features
+        assert(predictionJson.getJsonObject(0).get("features") == null);
+        
+        // Specify that we want to see features, then ensure that an empty feature object is included
+        prediction = prediction.withSignificantFeatures();
+        predictionJson = prediction.iterator().next().right.getJson();
+        assert(predictionJson.getJsonObject(0).get("features").toString().equals("{}"));
+    }
+    
+    @Test public void testNonTrivialRules() throws Exception {
+        String json = "{\"task\":{\"uuid\":\"00000000-0000-0000-0000-000000000000\",\"name\":\"ClassifyCats\"," +
```

Comment:
I'd like to see one additional test case here -- a task that just uses ClarabridgeRules, where one label is non-trivial.